app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt

```diff
@@ -77,15 +77,18 @@ public class MainActivity : AppCompatActivity(), MainViewController, RouteCallba
     var findMe: MapData? = null
     var searchResults: MapData? = null
 
-    var findMeButton: ImageButton? = null
-    var routePreviewView: RoutePreviewView? = null
-    var routeModeView: RouteModeView? = null
-    var reverseButton: ImageButton? = null
-    var viewListButton: Button? = null
-    var startNavigationButton: Button? = null
-    var byCar: RadioButton? = null
-    var byBike: RadioButton? = null
-    var byFoot: RadioButton? = null
+    val findMeButton: ImageButton by lazy { findViewById(R.id.find_me) as ImageButton }
+    val routePreviewView: RoutePreviewView by lazy { findViewById(R.id.route_preview) as RoutePreviewView }
+    val routeModeView: RouteModeView by lazy { findViewById(R.id.route_mode) as RouteModeView }
+    val reverseButton: ImageButton by lazy { findViewById(R.id.route_reverse) as ImageButton }
+    val viewListButton: Button by lazy { findViewById(R.id.view_list) as Button }
+    val startNavigationButton: Button by lazy { findViewById(R.id.start_navigation) as Button }
+    val byCar: RadioButton by lazy { findViewById(R.id.by_car) as RadioButton }
+    val byBike: RadioButton by lazy { findViewById(R.id.by_bike) as RadioButton }
+    val byFoot: RadioButton by lazy { findViewById(R.id.by_foot) as RadioButton }
+    val compass: CompassView by lazy { findViewById(R.id.compass_view) as CompassView }
+    val routePreviewCompass: CompassView by lazy { findViewById(R.id.route_preview_compass_view) as CompassView }
```

Comment:
Good point. In this case its not actually constructing the view lazily but rather doing a lookup to find it in the view hierarchy. The actual instance is created when the xml layout gets inflated.

The nice advantage of doing it this way is it delays the lookup until after the view is inflated so these properties don't have to be declared nullable.